packages/vhd-lib/Vhd/VhdNegative.js

```diff
@@ -37,11 +37,15 @@ class VhdNegative extends VhdAbstract {
     this.#parent = parent
     this.#child = child
 
-    assert.strictEqual(UUID.stringify(child.header.parentUuid), UUID.stringify(parent.footer.uuid), 'NOT_CHAINED')
-    assert.strictEqual(child.footer.diskType, DISK_TYPES.DIFFERENCING, 'CHILD_NOT_DIFFERENCING')
+    assert.strictEqual(
+      UUID.stringify(child.header.parentUuid),
+      UUID.stringify(parent.footer.uuid),
+      new Error('NOT_CHAINED')
```

Comment:
Other solution for this fix: https://github.com/vatesfr/xen-orchestra/commit/be05b69d41ff009875d3ca24d949b6b716f7836d

- avoids change in production code
- keep production code simple
- keep `AssertionError` extra info which can be useful.

Reply:
I like it, it's better than my proposition